mobify/platform-scaffold
# WEB-1468 & WEB-1505: Include prerender modals in DOM when not open

This PR updates the modal manager so it includes modals with the prerender flag in the DOM. So they can be crawled by search engines.


 **JIRA**: https://mobify.atlassian.net/browse/WEB-1505 & https://mobify.atlassian.net/browse/WEB-1486
 **Linked PRs**: https://github.com/mobify/progressive-web-sdk/pull/849

## Changes
- Add prerender flag to navigation modal component
- Get modals that need to be prerendered when modal manager initializes
- Update modal manager render function so it will include closed prerender modals in the DOM

## Todos

## How to test-drive this PR
- cd into the SDK branch seo-nav-changhes
- run `npm run prod:build`
- run `npm link`
- cd back to the scaffold
- run `npm link progressive-web-sdk`
- run `npm start`
- Preview merlins potions
- Inspect the modal manager container
- open and close the nav
- the modal manager container should always include the nav markup, when closed it will be within a div in the modal manager container
